### PR TITLE
feat(Armorgames): add buttons setting & perf

### PR DESCRIPTION
**Pull request by DarkVillager**
## Description 

Refactor & add settings for buttons
## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/A/Armorgames/presence.ts [hunk 1]
```diff
@@ -2,135 +2,87 @@ const presence = new Presence({
...
+		) {
+			presenceData.state = document.querySelector<HTMLInputElement>(
+				'[class="input-xlarge"]'
+			).value;
+			presenceData.buttons = [
+				{
+					label: "Browse Searches",
+					url: href,
+				},
+			];
```
Comment (EncryptedDev) [inline websites/A/Armorgames/presence.ts:28]: I would never see a case where someone would like have a direct link to the search url they are looking for.
```suggestion
```

File: websites/A/Armorgames/presence.ts [hunk 2]
```diff
@@ -2,135 +2,87 @@ const presence = new Presence({
...
+		} else if (search?.value) presenceData.state = search?.value;
+		else delete presenceData.details;
+
+		presenceData.smallImageKey = "searching";
+	} else if (pathname === "/") presenceData.details = "Viewing The Home page";
```
Comment (EncryptedDev) [inline websites/A/Armorgames/presence.ts:33]:
```suggestion
	} else if (pathname === "/") presenceData.details = "Viewing he home page";
```

File: websites/A/Armorgames/presence.ts [hunk 3]
```diff
@@ -2,135 +2,87 @@ const presence = new Presence({
...
-		presenceData.details = "Viewing Games Made By:";
-		presenceData.state = title.textContent;
-		presenceData.buttons = [
-			{ label: "View Profile", url: window.location.href },
-		];
-	} else if (page === "/register" || page === "/register/")
+	} else if (pathname === "/community")
+		presenceData.details = "Viewing the forums";
+	else if (pathname.includes("/forum/")) {
+		presenceData.details = "Viewing Forum";
```
Comment (EncryptedDev) [inline websites/A/Armorgames/presence.ts:47]:
```suggestion
		presenceData.details = "Viewing forum";
```

File: websites/A/Armorgames/presence.ts [hunk 4]
```diff
@@ -2,135 +2,87 @@ const presence = new Presence({
...
+		presenceData.state = document
+			.querySelector("#content-canvas > main > h1")
+			.textContent.replace("Forums → ", "");
+		presenceData.buttons = [{ label: "View Forum", url: href }];
+	} else if (pathname.includes("thread")) {
+		presenceData.details = "Reading Forum Thread";
```
Comment (EncryptedDev) [inline websites/A/Armorgames/presence.ts:53]:
```suggestion
		presenceData.details = "Reading a forum thread";
```

File: websites/A/Armorgames/presence.ts [hunk 5]
```diff
@@ -2,135 +2,87 @@ const presence = new Presence({
...
+		presenceData.state = document.querySelector("#thread-title").textContent;
+		presenceData.buttons = [{ label: "View Thread", url: href }];
+	} else if (pathname.includes("/user/")) {
+		presenceData.details = "Viewing the profile of";
```
Comment (EncryptedDev) [inline websites/A/Armorgames/presence.ts:57]:
```suggestion
		presenceData.details = "Viewing a user's profile";
```

File: websites/A/Armorgames/presence.ts [hunk 6]
```diff
@@ -2,135 +2,87 @@ const presence = new Presence({
...
+		presenceData.state = document.querySelector("#user-data > h1").textContent;
+		presenceData.buttons = [{ label: "View Profile", url: href }];
+	} else if (pathname.includes("/author/")) {
+		presenceData.details = "Viewing Games Made By";
```
Comment (EncryptedDev) [inline websites/A/Armorgames/presence.ts:61]:
```suggestion
		presenceData.details = "Viewing games created by";
```

File: websites/A/Armorgames/presence.ts (hunk 2 above)
Comment (RisingSunLight42) [reply websites/A/Armorgames/presence.ts:33]: Looks like you've edited to fast, a `t` is missing for the `the` !

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**